pkg/infra/spot.go

```diff
@@ -66,7 +66,7 @@ var _ = Describe("[Feature:Machines] Running on Spot", func() {
 		Expect(machines).To(HaveLen(3))
 
 		for _, machine := range machines {
-			Expect(machine.Spec.GetLabels()).To(HaveKeyWithValue(machinecontroller.MachineInterruptibleInstanceLabelName, ""))
+			Expect(machine.Spec.Labels).To(HaveKeyWithValue(machinecontroller.MachineInterruptibleInstanceLabelName, ""))
```

Comment:
why? does this make any difference?

Reply:
I realised this is because we use our own `ObjectMeta`.
@JoelSpeed do we still need that?
anyway can tackle separately so
/lgtm

Reply:
If we rely on webhooks for validation, we may be able to remove it now, will add a card to investigate switching back for a future sprint. It won't protect us completely because we have the failure policy open, so we could regress on the bug we fixed with it.